metakb/harvesters/civic.py

```diff
@@ -0,0 +1,377 @@
+        except:  # noqa: E722 # TODO: Add specific exception error
+            print('CIViC Harvester was not successful.')
+            return False
+
+    def _create_json(self, evidence, genes, variants, assertions):
+        """Create a composite json file containing evidence, genes, variants,
+        and assertions.
+        """
+        composite_dict = {
+            'evidence': evidence,
+            'genes': genes,
+            'variants': variants,
+            'assertions': assertions
+        }
+
+        # Create composite json
+        with open(f'{PROJECT_ROOT}/data/civic/civic_harvester.json',
+                  'w+') as f:
+            json.dump(composite_dict, f)
+            f.close()
+
+        # Create individual json for evidence, genes, variants, and assertions
+        data = ['evidence', 'genes', 'variants', 'assertions']
+        for d in data:
+            with open(f'{PROJECT_ROOT}/data/civic/{d}.json', 'w+') as f:
+                f.write(json.dumps(composite_dict[d]))
+                f.close()
+
+    def _harvest_evidence(self):
+        """Harvest evidence."""
+        evidence_classes = civicpy.get_all_evidence()
+        evidence = list()
+
+        for ev in evidence_classes:
+            ev_record = self._evidence_item(ev, is_evidence=True)
+            evidence.append(ev_record)
+        return evidence
+
+    def _harvest_genes(self):
+        """Harvest genes."""
+        genes = civicpy.get_all_genes()
+        genes_list = list()
+        for gene in genes:
+            g = self._harvest_gene(gene)
+            genes_list.append(g)
+        return genes_list
+
+    def _harvest_variants(self):
+        """Harvest variants."""
+        variants = civicpy.get_all_variants()
+        variants_list = list()
+
+        for variant in variants:
+            v = self._harvest_variant(variant)
+            variants_list.append(v)
+        return variants_list
+
+    def _harvest_assertions(self):
+        """Harvest assertions."""
+        assertions = civicpy.get_all_assertions()
+        assertions_list = list()
+
+        for assertion in assertions:
+            a = self._harvest_assertion(assertion)
+            assertions_list.append(a)
+        return assertions_list
+
+    def _harvest_gene(self, gene):
+        g = {
+            'id': gene.id,
+            'name': gene.name,
+            'entrez_id': gene.entrez_id,
+            'description': gene.description,
+            'variants': [
+                {
+                    'name': variant.name,
+                    'id': variant.id,
+                    'evidence_items_temp': [
+                        e.status for e in variant.evidence_items
+                    ]
+                }
+                for variant in gene.variants
+            ],
+            'aliases': gene.aliases,
+            # TODO: Add lifecycle_actions, sources
+            'type': gene.type
+        }
+
+        # Update evidence_items
+        for v in g['variants']:
```

Comment:
Assume we are initializing an aggregator here.   Intent unclear, will other harvesters do the same?

Reply:
This is just for CIViC harvesting genes. CIViCpy stored a list of strings for accepted, rejected, and submitted counts. This was to emulate the way CIViC API displayed evidence items which is a dictionary where the key is accepted, rejected, or submitted and the value is an int of the counts.